packages/story-utils/src/generateSnapshot.ts

```diff
@@ -0,0 +1,46 @@
+import { render, RenderResult } from '@testing-library/react';
+import { getStoriesFromStoryFileExports } from './getStories';
+import type { StoryData, StoryFileExports } from './getStories';
+import prepareStory from './prepareStory';
+
+type TestOptions<Args> = {
+  /**
+   * get the snapshot either synchronously, returning the ChildNode,
+   * or asynchronously returning a promise that resolves to the ChildNode.
+   */
+  getSnapshot?: (
+    wrapper: RenderResult,
+  ) => Promise<ChildNode | null> | ChildNode | null;
+  /**
+   * Override the args for the story. Useful for event handlers that may be missing.
+   */
+  argOverrides?: Partial<Args>;
+};
+
+function getDefaultSnapshot(wrapper: RenderResult) {
+  return wrapper.container.firstChild;
```

Comment:
Looks like the weird TypeScript errors are occuring because:
- We're including `@types/react` as a dependency
- But not configuring TypeScript to include the "DOM" lib

Fixed by adding DOM as a lib in tsconfig.base.json

```diff
- "lib": ["es2019"],
+ "lib": ["DOM", "es2019"],
```

Figure this out by going to the definition of `wrapper.container`, and then from there to the definition of `HTMLElement`.

Instead of going to the real definition for HTMLElement, it led me to a stub included by `@types/react`, which had this note:

```
/*
React projects that don't include the DOM library need these interfaces to compile.
React Native applications use React, but there is no DOM available. The JavaScript runtime
is ES6/ES2015 only. These definitions allow such projects to compile with only `--lib ES6`.

Warning: all of these interfaces are empty. If you want type definitions for various properties
(such as HTMLInputElement.prototype.value), you need to add `--lib DOM` (via command line or tsconfig.json).
*/
```

Reply:
Ahhh thank you! When I went to the def, it didn't lead me to the above 🤔